docs/concepts/unions.md

```diff
@@ -76,25 +76,59 @@ print(User(id='456'))  # (2)
 2. We're in lax mode and the numeric string `'123'` is valid as input to the first member of the union, `int`.
    Since that is tried first, we get the surprising result of `id` being an `int` instead of a `str`.
 
+!!! tip
+    For complex cases, it's recommended to use either [`left_to_right`](#left-to-right-mode) mode or [discriminated unions](#discriminated-unions).
+    With these approaches, you can control which member of the union to validate against, and in which order.
+    If you're looking for incredibly specialized behavior, you can use a [custom validator](../concepts/validators.md#field-validators).
+
 ### Smart Mode
 
 Because of the surprising side effects of `union_mode='left_to_right'`, in Pydantic >=2 the default mode for `Union` validation is `union_mode='smart'`.
 
-In this mode, pydantic scores a match of a union member into one of the following three groups (from highest score to lowest score):
+In this mode, pydantic selects the best match for the input from the union members based on two main factors:
```

Comment:
I suggest we rewrite this to say something like

```suggestion
In this mode, pydantic attempts to select the best match for the input from the union members. The exact algorithm may change between Pydantic minor releases to allow for improvements in both performance and accuracy.
```

Maybe then move the tip about complex cases / user control here too?